### PR TITLE
Review added signal in nodebb handlers [#LP-316]

### PR DESCRIPTION
**Pull request by rosheen33**
* Signals removed from models file to avoid conflicts
* **sync_user_info_with_nodebb** Signal updated to update NodeBB whenever **InterestsSurvey** is updated

connects [#LP-316](https://philanthropyu.atlassian.net/browse/LP-316)

### CODE REVIEW

File: common/djangoapps/nodebb/signals/handlers.py [hunk 1]
```diff
@@ -5,15 +5,56 @@
...
+@receiver(post_save, sender=UserInfoSurvey)
+@receiver(post_save, sender=ExtendedProfile)
+@receiver(post_save, sender=InterestsSurvey)
+def sync_user_info_with_nodebb(sender, instance, **kwargs):  # pylint: disable=unused-argument, invalid-name
+    """ Sync user information with  """
+    user = instance.user
+
+    if user:
+        try:
+            extended_profile = user.extended_profile
```
Comment (ziafazal) [inline common/djangoapps/nodebb/signals/handlers.py]: This is not efficient. we are hitting DB at-least 3 times for each save call on any of UserInfoSurvey, ExtendedProfile or InterestsSurvey. I think we update data based on sender without grabbing the data from these models again. something like 
```
        if sender == InterestsSurvey:
               data_to_sync = {
                     'interests': ' '.join([area.label for area in interest_survey.capacity_areas.all()])
              }
         elif sender == ExtendedProfile:
             data_to_sync = {
                 "first_name": extended_profile.first_name,
                 "last_name": extended_profile.last_name,
             }
         else:
             data_to_sync = {
                 "city_of_residence": user_info_survey.city_of_residence,
                 "country_of_residence": user_info_survey.country_of_residence
             }

```

File: common/djangoapps/nodebb/signals/handlers.py [hunk 2]
```diff
@@ -5,15 +5,56 @@
...
+                "last_name": extended_profile.last_name,
+                "city_of_residence": user_info_survey.city_of_residence,
+                "country_of_residence": user_info_survey.country_of_residence
+            }
+
+        status_code, response_body = NodeBBClient().users.update_profile(user.username, kwargs=data_to_sync)
+
+        if status_code != 200:
+            log.error(
+                "Error: Can not update user({}) on nodebb due to {}".format(user.username, response_body)
```
Comment (ziafazal) [inline common/djangoapps/nodebb/signals/handlers.py]: Unicode data is going to break this logging.

**Comment by rosheen33** [comment]
@ziafazal Kindly review again.

File: common/djangoapps/nodebb/signals/handlers.py [hunk 3]
```diff
@@ -24,6 +24,42 @@ def set_is_featured_false(sender, instance, created, **kwargs):
...
+    """ Sync user information with  """
+    user = instance.user
+
+    if user:
+        if sender == ExtendedProfile:
+            data_to_sync = {
+                "first_name": instance.first_name,
+                "last_name": instance.last_name,
+            }
+
```
Comment (ziafazal) [inline common/djangoapps/nodebb/signals/handlers.py]: unnecessary line break.

**Pull request merged**